bioio_ome_zarr/writers/ome_zarr_writer.py

```diff
@@ -0,0 +1,489 @@
+                )
+        else:
+            compressor = self.compressor
+
+        self.datasets = []
+
+        if self.zarr_format == 2:
+            # v2
+            if not self._chunk_shape_explicit:
+                # If 5D TCZYX, use legacy z-slice per-level chunking; otherwise
+                # keep the suggested per-level chunking already prepared.
+                is_tczyx = self.ndim == 5 and [n.lower() for n in self.axes.names] == [
+                    "t",
+                    "c",
+                    "z",
+                    "y",
+                    "x",
+                ]
+                if is_tczyx:
+                    self.chunk_shapes_per_level = compute_level_chunk_sizes_zslice(
+                        self.level_shapes
+                    )
+
+            for lvl, shape in enumerate(self.level_shapes):
+                chunks_lvl = self.chunk_shapes_per_level[lvl]
+                arr = self.root.zeros(
+                    name=str(lvl),
+                    shape=shape,
+                    chunks=chunks_lvl,
+                    dtype=self.dtype,
+                    compressor=compressor,
+                    zarr_format=2,
+                    dimension_separator="/",
+                )
+                self.datasets.append(arr)
+        else:
+            # v3
+            for lvl, shape in enumerate(self.level_shapes):
+                chunks_lvl = self.chunk_shapes_per_level[lvl]
+                kwargs: Dict[str, Any] = {
+                    "name": str(lvl),
+                    "shape": shape,
+                    "chunks": chunks_lvl,
+                    "dtype": self.dtype,
+                    "compressors": compressor,
+                    "chunk_key_encoding": {
+                        "name": "default",
+                        "separator": "/",
+                    },
+                }
+                if self.shard_factor is not None:
+                    kwargs["shards"] = tuple(
+                        c * f for c, f in zip(chunks_lvl, self.shard_factor)
+                    )
+                arr = self.root.create_array(**kwargs)
+                self.datasets.append(arr)
+
+        # Write metadata
+        self._write_metadata()
+        self._metadata_written = True
+
+        self._initialized = True
+
+    def _open_root(self) -> zarr.Group:
+        """Accept a path/URL or Store-like and return an opened root group."""
+        if isinstance(self.store, str):
+            if "://" in self.store:
+                fs = zarr.storage.FsspecStore(self.store, mode="w")
+                return zarr.open_group(store=fs, mode="w", zarr_format=self.zarr_format)
+            return zarr.open_group(self.store, mode="w", zarr_format=self.zarr_format)
+        return zarr.group(
+            store=self.store,
+            overwrite=True,
+            zarr_format=self.zarr_format,
+        )
+
+    def preview_metadata(self) -> Dict[str, Any]:
```

Comment:
Lovely! Should be moved to public interface since it is public even if we don't necessary care if anyone else uses it